data/transactions/logic/eval.go

```diff
@@ -1985,6 +1985,28 @@ func branchTarget(cx *EvalContext) (int, error) {
 	return target, nil
 }
 
+func branchSwitch(cx *EvalContext, branchIdx uint64) (int, uint64, error) {
+	numOffsets, bytesUsed := binary.Uvarint(cx.program[cx.pc+1:])
+	if numOffsets == 0 {
+		return 0, 0, fmt.Errorf("number of offsets must be greater than 0")
+	}
+	if branchIdx >= numOffsets {
+		return 0, 0, fmt.Errorf("provided branch index %d exceeds max offset index %d", branchIdx, numOffsets-1)
+	}
+
+	pos := uint64(cx.pc+1+bytesUsed) + (2 * branchIdx)
```

Comment:
Would be nice to set `end` to `cx.pc+1+bytesUsed` and use it in the next few lines as a way to make things clearer.